Refactor(eos_cli_config_gen)!: Allow for multiple radius-server with the same host to render both TLS and non-TLS server (#6287)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

+       source_interface: Management0
```

#### `radius_server.hosts` data-model changed to `radius_server.vrfs` data-model

In AVD 5.x, the data model `radius_server.hosts` could generate single command using the same VRF for a particular host and when TLS was configured, the `key` was ignored.

Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured and the TLS and `key` can be configured together.

```diff
radius_server:
-  hosts:

> -    - host: 10.10.10.157

Reviewer: Please add an example here for 
1. same host
2. TLS and `key` configured together

Author: Added

| `ip_ssh_client_source_interfaces` | `ip_ssh_client` |
| `ip_telnet_client_source_interfaces` | `ip_telnet_client` |
| `ip_tftp_client_source_interfaces` | `ip_tftp_client` |

> | `radius_server.hosts` | `radius_server.vrfs` |

Reviewer 2: ```suggestion
| `radius_server.hosts` | `radius_server.vrfs` and `radius_server.servers` |
```

+       name: MGMT
```

#### `radius_server.hosts` data-model changed to `radius_server.vrfs` data-model

> In AVD 5.x, the data model `radius_server.hosts` could generate single command using the same VRF for a particular host and when TLS was configured, the `key` was ignored.

Reviewer 2: Technically we can still only render either key or TLS - but with removing the primary key we allow user to configure several
```suggestion
In AVD 5.x, the data model `radius_server.hosts` could generate a single command using the same VRF for a particular host.
```

> Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured and the TLS and `key` can be configured together.

Reviewer 2: ```suggestion
Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured. It is now also possible to configure side-by-side TLS and non-TLS servers with the same host IP.
```

+       name: MGMT
```

#### `radius_server.hosts` data-model changed to `radius_server.vrfs` data-model

In AVD 5.x, the data model `radius_server.hosts` could generate a single command using the same VRF for a particular host.

Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured. It is now also possible to configure side-by-side TLS and non-TLS servers with the same host IP.

```diff
radius_server:
-  hosts:
-    - host: 10.10.10.157
-      vrf: mgmt
-      key: 071B245F5A
-    - host: 10.10.11.156
-      vrf: default
-      timeout: 1
-      retransmit: 1
-      key: 071B245F5A
-      tls:
-        enabled: true
-        port: 1700
-    - host: 10.10.10.156
-      vrf: default
-      timeout: 30
-      retransmit: 20
-      key: 071B245F5b
-    - host: 10.10.11.159
-      vrf: mgmt
-      retransmit: 1
-      key: 071B245F5A
+  vrfs:

> +    - name: default

Reviewer 2: apologies for just realising this now but the vrf `default` is not rendered right?

Author: it gets rendered like this -
```
radius-server host 10.10.11.156 tls port 1700 timeout 1 retransmit 1
```
without the `default ` keyword

Reviewer 3: we need to add a model to allow the servers without a VRF and use that for VRF default instead of checking the value and removing the VRF from the generated CLI. This can be done later so we just need a new issue on this.

Reviewer 3: If you have hosts here already we should keep that. And then just deprecate the VRF key under the hosts and point to the new place.

Reviewer 2: hosts wont do as it is called servers under vrfs today

Author: Updated the model

+       name: MGMT
```

#### `radius_server.hosts` data-model changed to `radius_server.vrfs` data-model

In AVD 5.x, the data model `radius_server.hosts` could generate a single command using the same VRF for a particular host.

> Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured. It is now also possible to configure side-by-side TLS and non-TLS servers with the same host IP.

Reviewer 2: ```suggestion
Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured and `radius_server.servers` for the default VRF. It is now also possible to configure side-by-side TLS and non-TLS servers with the same host IP.
```

```diff
radius_server:
-  hosts:
-    - host: 10.10.10.157
-      vrf: mgmt
-      key: 071B245F5A
-    - host: 10.10.11.156
-      vrf: default
-      timeout: 1
-      retransmit: 1
-      key: 071B245F5A
-      tls:
-        enabled: true
-        port: 1700
-    - host: 10.10.10.156
-      vrf: default
-      timeout: 30
-      retransmit: 20
-      key: 071B245F5b
-    - host: 10.10.11.159
-      vrf: mgmt
-      retransmit: 1
-      key: 071B245F5A
+  vrfs:

> +    - name: default

Reviewer 2: you need to move out the VRF default to your new model

Author: updated

+       name: MGMT
```

#### `radius_server.hosts` data-model changed to `radius_server.vrfs` data-model

In AVD 5.x, the data model `radius_server.hosts` could generate a single command using the same VRF for a particular host.

> Starting with AVD 6.0.0, `radius_server.hosts` has been replaced by `radius_server.vrfs`, where the VRF is now the primary key, under which multiple hosts can be configured and `radius_server.servers` for the default VRF. It is now also possible to configure side-by-side TLS and non-TLS servers with the same host IP.

Reviewer 4: Had `radius_server.hosts` been really replaced by `radius_server.vrfs`? And is `VRF is now the primary key` correct?
From the sema and examples it looks like `radius_server.hosts` is just now for `deafult` VRF and `radius_server.vrfs` is for all servers in non-default VRFs

Author: yes missed to update the porting guide after refactoring the model.
Updated now. Thanks.

```diff
radius_server:
-  hosts:
-    - host: 10.10.10.157
-      vrf: mgmt
-      key: 071B245F5A
-    - host: 10.10.11.156
-      vrf: default
-      timeout: 1
-      retransmit: 1
-      key: 071B245F5A
-      tls:
-        enabled: true
-        port: 1700
-    - host: 10.10.10.156
-      vrf: default
-      timeout: 30
-      retransmit: 20
-      key: 071B245F5b
-    - host: 10.10.11.159
-      vrf: mgmt
-      retransmit: 1
-      key: 071B245F5A
+  vrfs:
+    - name: default
+      servers:

> +        - host: 10.10.11.156

Reviewer 4: Current code renders this with `vrf default`:

`radius-server host 10.10.11.156 vrf default tls port 1700 timeout 1 retransmit 1`

Reviewer 2: which is not needed hence the new model to have a way to cleanly identify this

Author: Updated
